egs/wsj/s5/utils/format_lm_sri.sh

```diff
@@ -85,30 +85,15 @@ mkdir -p $out_dir
 cp -r $lang_dir/* $out_dir || exit 1;
 
 lm_base=$(basename $lm '.gz')
-gunzip -c $lm | utils/find_arpa_oovs.pl $out_dir/words.txt \
-  > $out_dir/oovs_${lm_base}.txt || exit 1;
-
-# Removing all "illegal" combinations of <s> and </s>, which are supposed to
-# occur only at being/end of utt.  These can cause determinization failures
-# of CLG [ends up being epsilon cycles].
-gunzip -c $lm \
-  | egrep -v '<s> <s>|</s> <s>|</s> </s>' \
-  | gzip -c > $tmpdir/lm.gz || exit 1;
-
 awk '{print $1}' $out_dir/words.txt > $tmpdir/voc || exit 1;
 
 # Change the LM vocabulary to be the intersection of the current LM vocabulary
 # and the set of words in the pronunciation lexicon. This also renormalizes the
 # LM by recomputing the backoff weights, and remove those ngrams whose
 # probabilities are lower than the backed-off estimates.
-change-lm-vocab -vocab $tmpdir/voc -lm $tmpdir/lm.gz -write-lm $tmpdir/out_lm \
-  $srilm_opts || exit 1;
-
-arpa2fst $tmpdir/out_lm | fstprint \
-  | utils/eps2disambig.pl | utils/s2eps.pl \
-  | fstcompile --isymbols=$out_dir/words.txt --osymbols=$out_dir/words.txt \
-    --keep_isymbols=false --keep_osymbols=false \
-  | fstrmepsilon | fstarcsort --sort_type=ilabel > $out_dir/G.fst || exit 1;
+change-lm-vocab -vocab $tmpdir/voc -lm $lm -write-lm - $srilm_opts | \
+  arpa2fst --disambig-symbol=#0 \
+           --read-symbol-table=$out_dir/words.txt - $out_dir/G.fst || exit 1
 
```

Comment:
Yes it would be good if you could test the resegmentation script too.  Bear
in mind that it could be broken for unrelated reasons.
Dan


On Tue, Apr 12, 2016 at 7:12 PM, Xiaohui Zhang <notifications@github.com>
wrote:

> In egs/wsj/s5/utils/format_lm_sri.sh
> <https://github.com/kaldi-asr/kaldi/pull/682#discussion_r59483974>:
>
> >  awk '{print $1}' $out_dir/words.txt > $tmpdir/voc || exit 1;
> >
> >  # Change the LM vocabulary to be the intersection of the current LM vocabulary
> >  # and the set of words in the pronunciation lexicon. This also renormalizes the
> >  # LM by recomputing the backoff weights, and remove those ngrams whose
> >  # probabilities are lower than the backed-off estimates.
> > -change-lm-vocab -vocab $tmpdir/voc -lm $tmpdir/lm.gz -write-lm $tmpdir/out_lm \
> > -  $srilm_opts || exit 1;
> > -
> > -arpa2fst $tmpdir/out_lm | fstprint \
> > -  | utils/eps2disambig.pl | utils/s2eps.pl \
> > -  | fstcompile --isymbols=$out_dir/words.txt --osymbols=$out_dir/words.txt \
> > -    --keep_isymbols=false --keep_osymbols=false \
> > -  | fstrmepsilon | fstarcsort --sort_type=ilabel > $out_dir/G.fst || exit 1;
> > +change-lm-vocab -vocab $tmpdir/voc -lm $lm -write-lm - $srilm_opts | \
> > +  arpa2fst --disambig-symbol=#0 \
> > +           --read-symbol-table=$out_dir/words.txt - $out_dir/G.fst || exit 1
> >
>
> Hi Dan I've tested on swbd. The decoding results till tri2 makes zero
> difference. It only involves utils/format_lm_sri.sh till tri2 stage though.
> steps/make_phone_graph.sh is involved in resegmentation. Shall I go ahead
> test it?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/682/files/6e8dd7a3e849f08d04b4fb28cd3fbd772297d4bd#r59483974>
>